app/uk/gov/hmrc/gform/models/FormModel.scala

```diff
@@ -302,7 +302,7 @@ case class FormModel[A <: PageMode](
       // User is trying to see invisible page, so we need to send him to appropriate SectionNumber instead
       availableSectionNumbers match {
         case Nil       => throw new IllegalArgumentException(s"Cannot find valid sectionNumber for $sectionNumber.")
-        case head :: _ => availableSectionNumbers.reverse.find(_ < sectionNumber).getOrElse(head)
+        case head :: _ => availableSectionNumbers.find(_ > sectionNumber).getOrElse(head)
```

Comment:
This will affect how navigation to invisible page is handled. Currently it goes to previous visible page (if one exists).
With this change it will go to next (possible never visited section).
Also with this change when you visit non existent section number bigger than last section number it navigates to first page instead of last visible page (as I'd expect).

There must be a better fix.

Or else we need to formalise how navigation to invisible pages should be handled.